subscribe.go

```diff
@@ -54,17 +56,65 @@ func (f HandlerFunc) Handle(msg SubMessage) error {
 
 // ----------------------------------------------------------------------------
 
+// SubStart defines starting position to consume messages.
+type SubStart int
+
+// SubStart options.
+const (
+	Newest SubStart = iota
+	Oldest
+)
+
+// SubOptions holds subscription options.
+type SubOptions struct {
+	// Start defines starting position to consume messages.
+	// Default: SubscriptionNewest
+	Start SubStart
```

Comment:
Probably, it could be `Default: implementation-specific`, but I wanted to keep it consistent across impls.

Though, it may make sense to do the impl-specific approach, as any use of `bps.InMem.Subscribe` or `bps/file.Subscriber.Subscribe` will ALWAYS require explicit `bps.Start(bps.Oldest)`.

Reply:
so yes, we should make this implementation specific, that's why I am not so sure about the `NewSubOptions(opts []SubOption)` approach as it takes away control over default from the implementations. It may be easier to have just a `struct SubOptions` which we optionally attach to `Subscribe` instead of the chained approach.

Reply:
I'm thinking about:

```go
func (s *SubImpl) Subscribe(..., options ...bps.Option) error {
  opts := (&bps.SubOptions{
    // impl-specific defaults
  }).Configure(options)
  ...
}
```

Still allows us to have a variadic option args (which I really like).

Will do it this way.